### ISSUE TITLE
Add remote API to extension manager

### ISSUE THREAD

**Issue by iainhallam** [splitbrain/dokuwiki#2538]
I find myself automating the installation of DokuWiki for several projects, and I can download and unpack it easily enough, then add configuration, but I haven't got a good way so far of installing preferred plugins. It's probably cleanest to add remote API calls for the extension manager's search, install and uninstall actions so that I could call them from an external script.

**Comment by splitbrain** [comment]
good idea. I would also like a cli component to be able to automate plugin management through salt or other configuration management tools. PR welcome.
